[NF-20097] Add the documentation for applicant details

zefta opened the pull request:
We're adding the documentation for the Applicant Details API endpoint, this endpoint allows you to edit the redirect_eligible column from the details table. Note that this endpoint is limited 1000 records per call.

Linked PR: https://github.com/Liaison-Intl/WebAdMIT/pull/7517

zefta commented on Gemfile.lock:
```diff
@@ -1,6 +1,6 @@
 GIT
   remote: https://github.com/Liaison-Intl/prmd.git
-  revision: 8cb419a9b12bbf3356fd904a0628679630a85dfd
+  revision: 20b43706b0421b02021f49a5658e2f38fadb3194
```
The gem has a bug which I fixed in https://github.com/Liaison-Intl/prmd/pull/1

Bunter commented on schemata/applicant_details.json:
```diff
@@ -0,0 +1,49 @@
+  "description": "The applicant details stores information about the applicant, only some of those information are editable.",
+  "links": [
+    {
+      "description": "Update a batch of Applicant Details, this endpoint is **limited to 1000 records per call**.",
```
We should probably say **limited to 1000 Applicants per call** ?
Details are 1:1 and I think it is a little clearer. what do you think?

dquimper replied:
👍

zefta commented:
@Bunter @dquimper I addressed your comments, let's not merge this until it's deployed to PROD otherwise the documentation will be live while the code is not deployed yet.

cdarne commented:
Indeed 😅
